packages/service-worker/src/low_level.ts

```diff
@@ -33,6 +33,22 @@ export interface UpdateActivatedEvent {
   current: {hash: string, appData?: Object};
 }
 
+/**
+ * An event emitted when the version of the app used by the service worker to serve this client is
+ * in a broken state that cannot be recovered from and a full page reload is required.
+ *
+ * For example, the ServiceWorker may not be able to retrieve a required resource, neither from the
+ * cache nor from the server. This could happen if a new version is deployed to the server and the
+ * ServiceWorker cache has been partially cleaned by the browser, removing some files of a previous
+ * app version but not all.
+ *
```

Comment:
I think it is worth adding a link to the [Service worker communication][1] guide (here and in the other events too). Something like:

```suggestion
 *
 * @see {@link guide/service-worker-communications Service worker communication guide}
 *
```

And it's also a good idea to add a similar annotation in the `SwUpdate` class jsdoc.
(Adding these links is a separate change, so feel free to make them on a new commit (either on this PR or a new one).)

[1]: https://angular.io/guide/service-worker-communications